packages/cli/src/commands/governance/upvote.ts

```diff
@@ -29,6 +30,43 @@ export default class Upvote extends BaseCommand {
       .runChecks()
 
     const account = await (await this.kit.contracts.getAccounts()).voteSignerToAccount(signer)
-    await displaySendTx('upvoteTx', await governance.upvote(id, account), {}, 'ProposalUpvoted')
+
+    const consideredProposals = await this.dequeueAllPossibleProposals(governance as any)
+
+    if (!consideredProposals.some((k) => k.id === id)) {
+      await displaySendTx('upvoteTx', await governance.upvote(id, account), {}, 'ProposalUpvoted')
```

Comment:
Would add a log statement that the tx were dequeued and no upvoting was needed/possible.

Reply:
Logging added